granulate_utils/config_feeder/client/bigdata/dataproc.py

```diff
@@ -0,0 +1,40 @@
+import json
+import subprocess
+from typing import Any, Dict, Optional
+
+from granulate_utils.config_feeder.core.models.node import NodeInfo
+
+
+def get_dataproc_node_info() -> Optional[NodeInfo]:
+    try:
+        metadata = _get_metadata()
+        attributes = metadata["attributes"]
+        properties = {
+            "region": attributes["dataproc-region"],
+            "cluster_name": attributes["dataproc-cluster-name"],
+        }
+        return NodeInfo(
+            external_id=str(metadata["id"]),
+            external_cluster_id=attributes["dataproc-cluster-uuid"],
+            is_master=attributes["dataproc-role"] == "Master",
+            provider="gcp",
+            bigdata_platform="dataproc",
+            properties=properties,
+        )
+    except (KeyError, json.JSONDecodeError):
```

Comment:
As continuation of: https://github.com/Granulate/granulate-utils/pull/158/files#r1242131736
What if the command will fail for example in any other reason (for example, no curl...)

Reply:
exception will be propagated to the client and ClientError will be raised.